### DISCUSSION TITLE
Random idea: rust version? For dynamic shape checking

### DISCUSSION THREAD

**Discussion by jaanaltosaar** [beartype/beartype#395] [Ideas]
In rust, for applications that use the runtime (e.g. WASMEdge, 
https://github.com/LlamaEdge/LlamaEdge/blob/main/README.md, etc), something like beartype would be useful.

I'm new to rust and am curious if anyone else is interested in this for webassembly/webgpu use cases.

Happy to chat if so! P.S. Not sure if this should be a discussion, feel free to move.

**Comment by leycec** [comment]
**...heh.** Big AI Healthcare Boss @jaanli with the hard-hitting query. Indeed, my bald-faced failure to port @beartype to Rust as `rustybeartype` *is* an issue if you want @beartype in Rust. However, my even balder-faced failure to bring @beartype itself close to completion in its native language is an even bigger issue if you want @beartype in Python.

There are many failures here. It's failure upon failure – compounding and ramifying one another. It's failure all the way down. Still, one of these failures is awfuller than the rest. I may affectionately pat Rust on the head on the weekends, <sup>*...bounds checker intensifies*</sup> but Python is still the nearest and dearest to my heart. I'll never cheat on you, Python! Never!

As you say, this kinda feels more like a discussion topic rather than an outstanding issue *per say.* With apologies, let's gently refactor this into a discussion for the good of @leycec's last remaining hair follicle.

> `rustybeartype`: "Why are you not a thing already?" :face_exhaling: 

![](https://media1.tenor.com/m/Dt5dAmBP5c8AAAAC/everybody-wake.gif)
<sup>Right: *@leycec doppelganger with hair.* Left: *Rust.*</sup>
